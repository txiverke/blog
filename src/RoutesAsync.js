import React from 'react'

import Loading from './components/Loading'

class RoutesAsync extends React.Component {
  state = {
    loaded: false
  }

  props: {
    props: mixed,
    loadingPromise: Promise<{ default: Class<React.Component<*, *, *>> }>
  }

  component = null

  async componentDidMount() {
    const module = await this.props.loadingPromise
    this.component = module.default
    this.setState({ loaded: true })
  }

  render() {
    if (this.state.loaded) {
      return <this.component {...this.props.props} />;
    }

    return <Loading />;
  }
}

export default RoutesAsync