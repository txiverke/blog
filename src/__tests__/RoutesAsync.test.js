// @flow

import React from 'react'
import { shallow } from 'enzyme'
import RoutesAsync from '../RoutesAsync'

describe('<RoutesAsync />', () => {
  it('should render itself', () => {
    const fakeComponent = () => <div>Just a fake component</div> 
    const fakePromise = { default: fakeComponent }
    const component = shallow(<RoutesAsync loadingPromise={fakePromise} />)
    expect(component).toMatchSnapshot()
  })
})