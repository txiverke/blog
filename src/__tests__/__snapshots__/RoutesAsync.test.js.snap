// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoutesAsync /> should render itself 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RoutesAsync
    loadingPromise={
        Object {
            "default": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Loader
        msg="Loading..."
/>,
      "className": "app-view app-view-centered",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "msg": "Loading...",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Loader
          msg="Loading..."
/>,
        "className": "app-view app-view-centered",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "msg": "Loading...",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
